base/sim/FairMCApplication.cxx

```diff
@@ -1338,4 +1291,29 @@ void FairMCApplication::UndoGeometryModifications()
     gGeoManager->ClearPhysicalNodes(kFALSE);
 }
 
-ClassImp(FairMCApplication);
+void FairMCApplication::ConstructSensitiveDetectors()
+{
+    std::map<std::string, FairModule*> cloneVolumeMap;
+
+    for (auto const& x : fMapSensitiveDetectors) {
+        std::string volName = x.first;   //.substr(0, x.first.find("#", 0));
+        if (volName.find('#') != std::string::npos) {
+            volName = volName.substr(0, volName.find("#", 0));
+            std::map<std::string, FairModule*>::iterator it;
+            it = cloneVolumeMap.find(volName);
+            LOG(debug) << "FairMCApplication::ConstructSensitiveDetectors got clone " << x.first << " " << x.second;
+            if (it != cloneVolumeMap.end())
+                continue;
+            cloneVolumeMap[volName] = x.second;
+            LOG(debug) << "FairMCApplication::ConstructSensitiveDetectors really do " << volName;
+        }
+        LOG(debug) << "FairMCApplication::ConstructSensitiveDetectors really do " << volName;
+        TVirtualMC::GetMC()->SetSensitiveDetector(volName, x.second);
+    }
+}
+
+void FairMCApplication::AddSensitiveModule(std::string volName, FairModule* module)
```

Comment:
Could you try a string_view here as well?
```suggestion
void FairMCApplication::AddSensitiveModule(std::string_view volName, FairModule* module)
```

Reply:
Same here:
```
FairMCApplication.cxx:1315:27: error: no viable overloaded operator[] for type 'std::map<std::string, FairModule *>' (aka 'map<basic_string<char, char_traits<char>, allocator<char>>, FairModule *>')
    fMapSensitiveDetectors[volName] = module;
    ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
```

Reply:
Thanks again for trying it!